spec.html

```diff
@@ -1585,158 +1585,121 @@ <h1>Number::bitwiseNOT ( _x_ )</h1>
-            </li>
-            <li>
-              Division of an infinity by a zero results in an infinity. The sign is determined by the rule already stated above.
-            </li>
-            <li>
-              Division of an infinity by a non-zero finite value results in a signed infinity. The sign is determined by the rule already stated above.
-            </li>
-            <li>
-              Division of a finite value by an infinity results in zero. The sign is determined by the rule already stated above.
-            </li>
-            <li>
-              Division of a zero by a zero results in *NaN*; division of zero by any other finite value results in zero, with the sign determined by the rule already stated above.
-            </li>
-            <li>
-              Division of a non-zero finite value by a zero results in a signed infinity. The sign is determined by the rule already stated above.
-            </li>
-            <li>
-              In the remaining cases, where neither an infinity, nor a zero, nor *NaN* is involved, the quotient is computed and rounded to the nearest representable value using IEEE 754-2019 roundTiesToEven mode. If the magnitude is too large to represent, the operation overflows; the result is then an infinity of appropriate sign. If the magnitude is too small to represent, the operation underflows and the result is a zero of the appropriate sign. The ECMAScript language requires support of gradual underflow as defined by IEEE 754-2019.
-            </li>
-          </ul>
+          <p>The abstract operation Number::divide takes arguments _x_ (a Number) and _y_ (a Number). It performs division, producing the quotient of _x_ and _y_ where _x_ is the dividend and _y_ is the divisor. It performs the following steps when called:</p>
+          <emu-alg>
+            1. If _x_ is *NaN* or _y_ is *NaN*, return *NaN*.
+            1. If _x_ is *+&infin;*<sub>𝔽</sub> or _x_ is *-&infin;*<sub>𝔽</sub>, then
+              1. If _y_ is *+&infin;*<sub>𝔽</sub> or _y_ is *-&infin;*<sub>𝔽</sub>, return *NaN*.
+              1. If _y_ is *+0*<sub>𝔽</sub> or _y_ is positive, return _x_.
+              1. If _x_ is *+&infin;*<sub>𝔽</sub>, return *-&infin;*<sub>𝔽</sub>. Otherwise, return *+&infin;*<sub>𝔽</sub>.
+            1. If _y_ is *+&infin;*<sub>𝔽</sub>, then
+              1. If _x_ is *+0*<sub>𝔽</sub> or _x_ is positive, return *+0*<sub>𝔽</sub>. Otherwise, return *-0*<sub>𝔽</sub>.
+            1. If _y_ is *-&infin;*<sub>𝔽</sub>, then
+              1. If _x_ is *+0*<sub>𝔽</sub> or _x_ is positive, return *-0*<sub>𝔽</sub>. Otherwise, return *+0*<sub>𝔽</sub>.
+            1. If _x_ is *+0*<sub>𝔽</sub> or _x_ is *-0*<sub>𝔽</sub>, then
+              1. If _y_ is *+0*<sub>𝔽</sub> or _y_ is *-0*<sub>𝔽</sub>, return *NaN*.
+              1. If _y_ is positive, return _x_.
+              1. If _x_ is *+0*<sub>𝔽</sub>, return *-0*<sub>𝔽</sub>. Otherwise, return *+0*<sub>𝔽</sub>.
```

Comment:
It's probably clearer to write `Return -_x_`. Ditto similar cases elsewhere; I'm going to stop commenting on all of them.